### DISCUSSION TITLE
SendMessage from React to Unity

### DISCUSSION THREAD

**Discussion by angela318** [jeffreylanters/react-unity-webgl#344] [Help Wanted!] [answered]
I am trying to add the capability to pause the unity game from website. 
So I added a method in one of the game object and when the pause button in website is clicked, it will sendMessage to Unity. 
But I receive the following error in console: 
SendMessage: object Spawner does not have receiver for function PauseGame!
I added public method PauseGame in the class as below. I don't know what else should I do. 
Any help is welcome, thanks 

This is the class: 
public class MonsterSpawner : MonoBehaviour
{

    [SerializeField]
    private GameObject[] monsterReference;

    private GameObject spawnedMonster;

    [SerializeField]
    private Transform leftPos, rightPos;

    private int randomIndex;
    private int randomSide;

    // Start is called before the first frame update
    void Start()
    {
        StartCoroutine(SpawnMonsters());
    }

    public void PauseGame() {
        Time.timeScale = 0;
    }
    public void ResumeGame() {  
        Time.timeScale = 1;
    }
    IEnumerator SpawnMonsters() {

        while (true) {

            yield return new WaitForSeconds(Random.Range(1, 5));

            randomIndex = Random.Range(0, monsterReference.Length);
            randomSide = Random.Range(0, 2);

            spawnedMonster = Instantiate(monsterReference[randomIndex]);

            // left side
            if (randomSide == 0)
            {

                spawnedMonster.transform.position = leftPos.position;
                spawnedMonster.GetComponent<Monster>().speed = Random.Range(4, 10);

            }
            else
            {
                // right side
                spawnedMonster.transform.position = rightPos.position;
                spawnedMonster.GetComponent<Monster>().speed = -Random.Range(4, 10);
                spawnedMonster.transform.localScale = new Vector3(-1f, 1f, 1f);

            }

        } // while loop

    }

} // class

**Comment by angela318** [comment, marked as answer]
This morning I try again and it works now :-) 
Thanks for the wonderful library. 
Sorry that I can't delete the post

**Reply by jeffreylanters** [reply]
No worries, I'm glad you've figured it out. Happy coding!

**Comment by ghost** [comment]
```SendMessage: object === User Data Manager does not have receiver for function ```
I still got this error but don't know how to fix @@
